### ISSUE TITLE
Fix warnings from tests

### ISSUE THREAD

**Issue by jakobatgithub** [jakobatgithub/TicTacToe#43]
tests/test_symmetric_matrix.py:198
  /Users/jakob/TicTacToe/tests/test_symmetric_matrix.py:198: PytestCollectionWarning: cannot collect test class 'TestMatrix' because it has a __init__ constructor (from: tests/test_symmetric_matrix.py)
    class TestMatrix(BaseMatrix):

tests/test_deep_q_agent.py::TestDeepQLearningAgent::test_symmetrized_loss
  /Users/jakob/TicTacToe/.venv/lib/python3.13/site-packages/torch/nn/modules/loss.py:610: UserWarning: Using a target size (torch.Size([1, 1])) that is different to the input size (torch.Size([1])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
    return F.mse_loss(input, target, reduction=self.reduction)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
